### ISSUE TITLE
Printer (compact): invalid generated code for prefix unary operators

### ISSUE THREAD

**Issue by DmitrySoshnikov** [babel/babel#1835]
Since parens aren't handled, printer (compact mode) generates invalid code for the following snippet we have in our code base:

```javascript
return (
  ((60 * +end_pieces[0]) + (+end_pieces[1])) -
  ((60 * +start_pieces[0]) + (+start_pieces[1]))
);
```

Generated:

```javascript
return (
 60*+end_pieces[0]++end_pieces[1]-(
 60*+start_pieces[0]++start_pieces[1]));
```

**Comment by DmitrySoshnikov** [comment]
Thanks!

**Comment by sebmck** [comment]
Fixed as of 5.6.8, thanks! Created issue #1837 to add smoke tests to ensure this and similar issues don't happen again.
